src/FsCheck/Gen.fs

```diff
@@ -313,6 +313,19 @@ module Gen =
     [<CompiledName("ListOf")>]
     let listOfLength n arb = sequence [ for _ in 1..n -> arb ]
 
+    ///Generates a random permutation of the given sequence.
+    //[category: Creating generators]
+    [<CompiledName("Shuffle")>]
+    let shuffle xs =
+        gen {
+            let g = choose (-922337203, 922337203)
```

Comment:
The generator used in the [Haskell QuickCheck version of `shuffle`](https://github.com/nick8325/quickcheck/commit/6c360c1b594b0243e51bfcd4e2428882570e7a51) is

```
choose (minBound :: Int, maxBound)
```

Where `mindBound :: Int` and `maxBound :: Int` values are:

```
λ> minBound :: Int
-9223372036854775808

λ> maxBound :: Int
9223372036854775807
```

---

If I use something like `int.MinValue` and `int.MaxValue` the Random algorithm throws an exception:

```
---- System.DivideByZeroException : Attempted to divide by zero.

----- Inner Stack Trace -----
..src\FsCheck\Random.fs(82,0): at FsCheck.Random.stdRange(Int32 l, Int32 h, StdGen rng)
..src\FsCheck\Random.fs(84,0): at FsCheck.Random.range@84-1.Invoke(StdGen rng)
..src\FsCheck\Gen.fs(163,0): at FsCheck.Gen.Choose@163.Invoke(Int32 _arg1, StdGen r)
..src\FsCheck\Gen.fs(295,0): at FsCheck.Gen.go@290-1T.Invoke(FSharpList`1 gs, ..)
..src\FsCheck\Gen.fs(63,0): at FsCheck.GenBuilder.bind@62.Invoke(Int32 n, StdGen r0)
..src\FsCheck\Gen.fs(157,0): at FsCheck.Gen.sample@156.Invoke(Int32 i, StdGen seed, ..)
..tests\FsCheck.Test\Helpers.fs(11,0): at FsCheck.Test.Helpers.sample@11.Invoke(Gen`1 gn)
..tests\FsCheck.Test\Gen.fs(126,0):
    at FsCheck.Test.Gen.Shuffle@125.Invoke(Unit unitVar)
    at System.Lazy`1.CreateValue()
    at System.Lazy`1.LazyInitValue()
    at System.Lazy`1.get_Value()
..src\FsCheck\Testable.fs(145,0): at FsCheck.Testable.Prop.safeForce[a](Lazy`1 body)
```

Perhaps this is a bug with the current Random implementation.